aea/launcher.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the implementation of multiple AEA configs launcher."""
+import multiprocessing
+from concurrent.futures.process import ProcessPoolExecutor
+from multiprocessing.synchronize import Event
+from os import PathLike
+from threading import Thread
+from typing import Awaitable, Dict, List, Optional, Sequence, Type
+
+from aea.aea import AEA
+from aea.aea_builder import AEABuilder
+from aea.exceptions import AEAException
+from aea.helpers.base import cd
+from aea.runner import AsyncExecutor, BaseAEAExecutor, ThreadExecutor
+
+
+def load_agent(agent_dir: PathLike) -> AEA:
+    """
+    Load AEA from directory.
+
+    :param agent_dir: agent configuration directory
+
+    :return: AEA instance
+    """
+    with cd(agent_dir):
+        return AEABuilder.from_aea_project(".").build()
+
+
+def _run_agent(agent_dir: PathLike, stop_event: Event) -> None:
+    """
+    Load and run agent in a dedicated process.
+
+    :param agent_dir: agent configuration directory
+    :param stop_event: multithreading Event to stop agent run.
+
+    :return: None
+    """
+    agent = load_agent(agent_dir)
+
+    def stop_event_thread():
+        stop_event.wait()
+        agent.stop()
+
+    Thread(target=stop_event_thread, daemon=True).start()
+    try:
+        agent.start()
+    except Exception as e:
+        exc = AEAException(f"Raised {type(e)}({e})")
+        exc.__traceback__ = e.__traceback__
+        raise exc
+    finally:
+        stop_event.set()
+
+
+class ProcessExecutor(BaseAEAExecutor):
+    """
+    AEA executor uses subprocess to run multiple AEAs.
+
+    Support only AEA direcories, not instances.
+    """
+
+    def __init__(self, agent_dirs: Sequence[PathLike]) -> None:
+        """
+        Init executor.
+
+        :param agents: sequence of agents dirs to run.
+        """
+        super().__init__(agent_dirs)  # type: ignore # tricky part, cause need to pass str instead of AEA
+        self._agents: Sequence[PathLike] = agent_dirs  # type: ignore # tricky part, cause need to pass str instead of AEA
```

Comment:
Is it really worth inheriting from BaseAEAExecutor? It seems this is conceptually not correct.

Reply:
There should be a `BaseAEAProjectExecutor` and a `BaseAEAInstanceExecutor`?

Reply:
it's a point.
process executor is quite similar to async and threaded executors, only one difference, that it accepts path instead of aea instance

need to find out how to refactor it better